File: client/src/dataModels/cptCode.ts
import { BaseEntityModel } from "./baseEntityModel";

export class CptCode extends BaseEntityModel {
    Code: string;
    Name: string;
    Description: string;

    constructor(id: string = "", isDelete: boolean = false) {
        super(id, isDelete);

        this.Code = "";
        this.Name = "";
        this.Description = "";
    }

    createFromEntityModel(entity: any) {
        const cptCode =
            new CptCode(entity.Id, !!entity.isDelete);

        cptCode.Code = entity.Code;
        cptCode.Name = entity.Name;
        cptCode.Description = entity.Description;

        return cptCode;
    }
}

File: api/Medico.Api/DB/Models/CptCode.cs
namespace Medico.Api.DB.Models
{
    public abstract class BaseCode : BaseEntity
    {
        public string Code { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }
    }

    public class CptCode : BaseCode
    {
    }
}
